src/objects/Button.ts

```diff
@@ -0,0 +1,65 @@
+import * as Phaser from 'phaser';
+
+type ButtonOptionsType = {
+  content: string;
+  width: number;
+  onClick: () => void;
+};
+
+class Button extends Phaser.GameObjects.Container {
+  public button: Phaser.GameObjects.NineSlice;
+
+  static preload(scene: Phaser.Scene) {
+    scene.load.atlas(
+      'ui',
+      'https://labs.phaser.io/assets/ui/nine-slice.png',
+      'https://labs.phaser.io/assets/ui/nine-slice.json',
+    );
+  }
+
+  constructor(
+    scene: Phaser.Scene,
+    x: number,
+    y: number,
+    options: ButtonOptionsType,
+  ) {
+    super(scene, x, y);
+
+    this.scene = scene;
+
+    const { content, width, onClick } = options;
+
+    this.button = scene.add.nineslice(
+      x,
+      y,
+      'ui',
+      'button-bg',
+      128,
+      110,
+      64,
+      64,
+    );
+
```

Comment:
Would setting origin here fix the offset collision box? this.button.setOrigin(0, 0); seems to fix the collision offset but the text is still off and sure we want the origin point at half hight/width 🤔

Reply:
ill give it a go!

Reply:
fixed on branch